### DISCUSSION TITLE
using labels or pvc

### DISCUSSION THREAD

**Discussion by 5cat** [datashim-io/datashim#218] [Q&A] [answered]
Why should I use [labels](https://github.com/datashim-io/datashim/blob/master/examples/templates/example-pod.yaml) instead of the kubernetic way [pvc](https://github.com/datashim-io/datashim/blob/master/examples/templates/example-pod-directpvc.yaml). Is the sole reason [simplicity](https://github.com/datashim-io/datashim/pull/168#issuecomment-1109592696)?

**Comment by srikumar003** [comment, marked as answer]
Hi, we introduced labels for two reasons: 1) It makes pod declarations simpler; and 2) It enables different uses of datasets e.g. as environment variables, instead of mounting alone. You can use the PVC, if you prefer.  But, in a future update, datashim may not generate PVCs with names same as datasets to cater for different use-cases e.g. a dataset that encompasses multiple data sources.

**Reply by 5cat** [reply]
I see, ok I will use labels for future compatibility. Thank you for the heads up.

But wouldint `annotations` be a better way instead of `labels`? an `datashim.io/dataset/0/name: example-dataset` be better? similar to other kubernetes applications such as traefik nginx and external-dns, they use annotations instead of labels.

**Reply by srikumar003** [reply]
You're welcome! We are exploring annotations but so far we do not have a clear idea of their advantages over labels for datasets.
